src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Serialization
+{
+    /// <summary>
+    /// Represents a <see cref="SourceText"/> which can be serialized for sending to another process. The text is not
+    /// required to be a live object in the current process, and can instead be held in temporary storage accessible by
+    /// both processes.
+    /// </summary>
+    internal sealed class SerializableSourceText
+    {
+        /// <summary>
+        /// The storage location for <see cref="SourceText"/>.
+        /// </summary>
+        /// <remarks>
+        /// At least one of <see cref="Storage"/> or <see cref="Text"/> will be non-<see langword="null"/>. If both are
+        /// provided, they must reference the same source text.
```

Comment:
```suggestion
        /// At least one of <see cref="Storage"/> or <see cref="Text"/> will be non-<see langword="null"/>. If both are
        /// provided, they must reference the same source text.
```

i don't see who ever ends up providing both.  so the doc comment right now doesn't make sense to me. it seems to describe something that can't happen, and i'd rather it describe only the things that can.

note: you could have this comment if your private constructor ever became public.  Or you could have a comment that says "if `private SerializableSourceText(ITemporaryTextStorageWithName? storage, SourceText? text)` were to become public, then it would be ok to provide both of these as long as blah blah was true"

Reply:
In what scenario would this restriction change the decisions made by a consumer of this type, using the current API provided by the type?

Reply:
➡️ Contract strengthened